app/models/katello/content.rb

```diff
@@ -28,7 +28,7 @@ def repositories
     end
 
     def redhat?
-      self.products.first.try(:redhat?)
+      products.redhat.any?
```

Comment:
When looking around the codebase I found the opportunity to clean up some of the determinations around upstream / redhat subscriptions